node.gyp

```diff
@@ -933,43 +1070,37 @@
         '<(SHARED_INTERMEDIATE_DIR)', # for node_natives.h
       ],
 
-      'defines': [ 'NODE_WANT_INTERNALS=1' ],
+      'defines': [ 'NODE_WANT_INTERNALS=1', '<@(var_defines)' ],
 
       'sources': [
+        'test/cctest/node_module_reg.cc',
         'test/cctest/node_test_fixture.cc',
         'test/cctest/test_aliased_buffer.cc',
         'test/cctest/test_base64.cc',
         'test/cctest/test_node_postmortem_metadata.cc',
         'test/cctest/test_environment.cc',
         'test/cctest/test_platform.cc',
         'test/cctest/test_util.cc',
-        'test/cctest/test_url.cc'
+        'test/cctest/test_url.cc',
       ],
 
       'conditions': [
-        [ 'node_use_openssl=="true"', {
-          'defines': [
-            'HAVE_OPENSSL=1',
-          ],
-        }],
-        [ 'node_use_perfctr=="true"', {
-          'defines': [ 'HAVE_PERFCTR=1' ],
-        }],
         ['v8_enable_inspector==1', {
           'sources': [
             'test/cctest/test_inspector_socket.cc',
-            'test/cctest/test_inspector_socket_server.cc'
-          ],
-          'defines': [
-            'HAVE_INSPECTOR=1',
+            'test/cctest/test_inspector_socket_server.cc',
```

Comment:
Why can't we merge these lists?

Reply:
merge? can you elaborate it more?

Reply:
are you talking about merging this into `variables`? I don't know `sources` is available inside `variables`. If yes, I will merge them.

Reply:
@refack I can't access `sources` inside `variables`. then what do you mean by merge?

Reply:
It's just that having the condition in multiple point makes it hard to grok.

So maybe define a conditionally filled variable
```python
  'node_inspector_sources' : [],
  ...
  ['v8_enable_inspector==1', {
    'node_inspector_sources' : [
      'test/cctest/test_inspector_socket.cc',
      'test/cctest/test_inspector_socket_server.cc',
      '<@(node_inspector_generated_sources)',
  ]
```

Then here (or anywhere else where `sources` if appended) just have an unconditional
```python
    'sources': [ '>@(node_inspector_sources)']
```

Reply:
I see. I can merge them.

Reply:
@refack I found gyp throws an error when the variable is an empty array. Therefore, I need to preload the `sources` in cctest target into that new variable: `var_sources`. Same as `var_defines`.